products/bridge/smart-contracts/contracts/core/Relayer.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.20;
+
+interface IRelayerEvents {
+    event Relayed(
+        uint indexed targetChainId,
+        address target,
+        bytes call,
+        uint gasLimit,
+        uint nonce
+    );
+}
+
+interface IRelayerErrors {
+    error FailedDeploymentInitialization();
+}
+
+interface IRelayer is IRelayerEvents, IRelayerErrors {
+    function nonces(address target) external returns (uint);
+
+    function relay(
+        uint targetChainId,
+        address target,
+        bytes calldata call,
+        uint gasLimit
+    ) external returns (uint);
+}
+
+contract Relayer is IRelayer {
+    // target => nonce
+    mapping(address => uint) public nonces;
```

Comment:
Since this isn't global yet I think this will break nonce checking in `Dispatcher` (which already implemented global nonce checking) - you're probably already aware of this but bringing this up just in case